src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -189,7 +189,11 @@ static bool CanConvertSectionForSwitchExpression(bool supportsOrPattern, Analyze
             var syntaxFactsService = document.GetRequiredLanguageService<ISyntaxFactsService>();
 
             // Get all the descendant if statements to process.
-            var ifStatements = editor.OriginalRoot.DescendantNodes().OfType<TIfStatementSyntax>();
+            // NOTE: We need to realize the nodes with 'ToArray' call here
+            // to ensure we strongly hold onto the nodes so that 'TrackNodes'
+            // invoked below, which does tracking based off a ConditionalWeakTable,
+            // tracks the nodes for the entire duration of this method.
```

Comment:
💭 I think there is a simpler explanation here: `DescendantNodes()` can be an expensive method, and we iterate more than once over the resulting enumerable.